rust/src/dataset/transaction.rs

```diff
@@ -397,6 +400,47 @@ impl Transaction {
 
         Ok((manifest, final_indices))
     }
+
+    fn handle_rewrite_fragments(
+        final_fragments: &mut Vec<Fragment>,
+        groups: &[RewriteGroup],
+        mut fragment_id: u64,
+    ) -> Result<()> {
+        for group in groups {
+            // If the old fragments are contiguous, find the range
+            let replace_range = {
+                let start = final_fragments.iter().enumerate().find(|(_, f)| f.id == group.old_fragments[0].id)
+                    .ok_or_else(|| Error::Internal { message:
```

Comment:
I'm not sure this is neccesarily an internal error.  It could happen, for example, if a delete was issued after the rewrite had already started.  Perhaps we need a new error for "unrecoverable conflict"

Reply:
That's a good point. I'll adjust this error.